sdk/eventhub/azure-eventhubs/azure/eventhub/eventprocessor/event_processor.py

```diff
@@ -0,0 +1,171 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# -----------------------------------------------------------------------------------
+
+from typing import Callable
+import uuid
+import asyncio
+import logging
+
+from azure.eventhub import EventPosition, EventHubError
+from azure.eventhub.aio import EventHubClient
+from ._cancellation_token import CancellationToken
+from .checkpoint_manager import CheckpointManager
+from .partition_manager import PartitionManager
+from .partition_processor import PartitionProcessor
+from .close_reason import CloseReason
+
+log = logging.getLogger(__name__)
+
+
+class EventProcessor(object):
+    def __init__(self, consumer_group_name: str, eventhub_client: EventHubClient,
```

Comment:
I find the order of parameters somewhat odd - I'd expect the client to be the first parameter. Also, would it make sense to make the client optional and allow for the creation of a client inside this method if one is not provided?

Reply:
Moved client to the first parameter.
To create a client requires other parameters. If we make client optional and let the creation of a client inside this method, this method will need to have all parameters of EventHubClient.__init__() parameters.
We can consider this in preview 3.

Reply:
This is where `**kwargs` come in handy :)